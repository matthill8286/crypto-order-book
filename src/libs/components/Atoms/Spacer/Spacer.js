function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { media } from "../../../styles/media";
import { css, styled } from "../../../styles/styled";
var StyledSpacer = styled.div.withConfig({
  displayName: "Spacer__StyledSpacer",
  componentId: "sc-66xpbd-0"
})(["display:block;width:100%;", ""], function (_ref) {
  var direction = _ref.direction,
      size = _ref.size,
      theme = _ref.theme;
  return direction === 'vertical' ? css(["width:", ";"], theme.spacing.baseMobile[size]) : css(["height:", ";", "{height:", ";}"], theme.spacing.baseMobile[size], media.md, theme.spacing.base[size]);
});
export var Spacer = function Spacer(_ref2) {
  var size = _ref2.size,
      _ref2$direction = _ref2.direction,
      direction = _ref2$direction === void 0 ? 'horizontal' : _ref2$direction,
      otherProps = _objectWithoutProperties(_ref2, ["size", "direction"]);

  return size ? React.createElement(StyledSpacer, _extends({
    size: size,
    direction: direction
  }, otherProps)) : null;
};
//# sourceMappingURL=Spacer.js.map