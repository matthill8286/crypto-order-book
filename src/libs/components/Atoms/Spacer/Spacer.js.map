{"version":3,"sources":["../../../../src/components/Atoms/Spacer/Spacer.tsx"],"names":["React","media","css","styled","StyledSpacer","div","direction","size","theme","spacing","baseMobile","md","base","Spacer","otherProps"],"mappings":";;;;;;AAAA,OAAOA,KAAP,MAAkB,OAAlB;AACA,SAASC,KAAT;AACA,SAASC,GAAT,EAAcC,MAAd;AAcA,IAAMC,YAAY,GAAGD,MAAM,CAACE,GAAV;AAAA;AAAA;AAAA,sCAId;AAAA,MAAGC,SAAH,QAAGA,SAAH;AAAA,MAAcC,IAAd,QAAcA,IAAd;AAAA,MAAoBC,KAApB,QAAoBA,KAApB;AAAA,SACAF,SAAS,KAAK,UAAd,GACIJ,GADJ,kBAEeM,KAAK,CAACC,OAAN,CAAcC,UAAd,CAAyBH,IAAzB,CAFf,IAIIL,GAJJ,qCAKgBM,KAAK,CAACC,OAAN,CAAcC,UAAd,CAAyBH,IAAzB,CALhB,EAOQN,KAAK,CAACU,EAPd,EAQkBH,KAAK,CAACC,OAAN,CAAcG,IAAd,CAAmBL,IAAnB,CARlB,CADA;AAAA,CAJc,CAAlB;AAkBA,OAAO,IAAMM,MAA6B,GAAG,SAAhCA,MAAgC,QAIvC;AAAA,MAHJN,IAGI,SAHJA,IAGI;AAAA,8BAFJD,SAEI;AAAA,MAFJA,SAEI,gCAFQ,YAER;AAAA,MADDQ,UACC;;AACJ,SAAOP,IAAI,GAAG,oBAAC,YAAD;AAAc,IAAA,IAAI,EAAEA,IAApB;AAA0B,IAAA,SAAS,EAAED;AAArC,KAAoDQ,UAApD,EAAH,GAAwE,IAAnF;AACD,CANM","sourcesContent":["import React from 'react'\nimport { media } from '@/styles/media'\nimport { css, styled } from '@/styles/styled'\n\nexport type SpacerDirection = 'horizontal' | 'vertical'\n\ninterface SpacerProps {\n  size: string\n  direction?: SpacerDirection\n}\n\ninterface StyledSpacerProps {\n  size: string\n  direction: SpacerDirection\n}\n\nconst StyledSpacer = styled.div<StyledSpacerProps>`\n  display: block;\n  width: 100%;\n\n  ${({ direction, size, theme }) =>\n    direction === 'vertical'\n      ? css`\n          width: ${theme.spacing.baseMobile[size]};\n        `\n      : css`\n          height: ${theme.spacing.baseMobile[size]};\n\n          ${media.md} {\n            height: ${theme.spacing.base[size]};\n          }\n        `}\n`\n\nexport const Spacer: React.FC<SpacerProps> = ({\n  size,\n  direction = 'horizontal',\n  ...otherProps\n}) => {\n  return size ? <StyledSpacer size={size} direction={direction} {...otherProps} /> : null\n}\n"],"file":"Spacer.js"}