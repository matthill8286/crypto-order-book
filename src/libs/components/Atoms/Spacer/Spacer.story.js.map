{"version":3,"sources":["../../../../src/components/Atoms/Spacer/Spacer.story.tsx"],"names":["storiesOf","React","FlexBox","styled","Spacer","Readme","StyledBlock","div","StyledVerticalBlock","module","add","info"],"mappings":"AAAA,SAASA,SAAT,QAA0B,kBAA1B;AACA,OAAOC,KAAP,MAAkB,OAAlB;AACA,SAASC,OAAT;AACA,SAASC,MAAT;AACA,SAASC,MAAT;AACA,OAAOC,MAAP;AAEA,IAAMC,WAAW,GAAGH,MAAM,CAACI,GAAV;AAAA;AAAA;AAAA,yDAAjB;AAOA,IAAMC,mBAAmB,GAAGL,MAAM,CAACI,GAAV;AAAA;AAAA;AAAA,0DAAzB;AAOAP,SAAS,CAAC,4BAAD,EAA+BS,MAA/B,CAAT,CACGC,GADH,CAEI,SAFJ,EAGI;AAAA,SACE,0CACE,oBAAC,WAAD,OADF,EAEE,oBAAC,MAAD;AAAQ,IAAA,IAAI,EAAE;AAAd,IAFF,EAGE,oBAAC,WAAD,OAHF,EAIE,oBAAC,MAAD;AAAQ,IAAA,IAAI,EAAE;AAAd,IAJF,EAKE,oBAAC,WAAD,OALF,CADF;AAAA,CAHJ,EAYI;AACEC,EAAAA,IAAI,EAAEN;AADR,CAZJ,EAgBGK,GAhBH,CAiBI,UAjBJ,EAkBI;AAAA,SACE,oBAAC,OAAD;AAAS,IAAA,aAAa,EAAC;AAAvB,KACE,oBAAC,mBAAD,OADF,EAEE,oBAAC,MAAD;AAAQ,IAAA,SAAS,EAAC,UAAlB;AAA6B,IAAA,IAAI,EAAE;AAAnC,IAFF,EAGE,oBAAC,mBAAD,OAHF,EAIE,oBAAC,MAAD;AAAQ,IAAA,SAAS,EAAC,UAAlB;AAA6B,IAAA,IAAI,EAAE;AAAnC,IAJF,EAKE,oBAAC,mBAAD,OALF,CADF;AAAA,CAlBJ,EA2BI;AACEC,EAAAA,IAAI,EAAEN;AADR,CA3BJ","sourcesContent":["import { storiesOf } from '@storybook/react'\nimport React from 'react'\nimport { FlexBox } from '@/components/Helper/FlexBox'\nimport { styled } from '@/styles/styled'\nimport { Spacer } from './Spacer'\nimport Readme from './Spacer.readme.md'\n\nconst StyledBlock = styled.div`\n  height: 20px;\n  width: 100%;\n  background: red;\n  float: left;\n`\n\nconst StyledVerticalBlock = styled.div`\n  height: 100px;\n  width: 20px;\n  background: red;\n  float: left;\n`\n\nstoriesOf('Design System/Atoms/Spacer', module)\n  .add(\n    'Default',\n    () => (\n      <>\n        <StyledBlock />\n        <Spacer size={'md'} />\n        <StyledBlock />\n        <Spacer size={'xxxl'} />\n        <StyledBlock />\n      </>\n    ),\n    {\n      info: Readme,\n    }\n  )\n  .add(\n    'Vertical',\n    () => (\n      <FlexBox flexDirection=\"row\">\n        <StyledVerticalBlock />\n        <Spacer direction=\"vertical\" size={'md'} />\n        <StyledVerticalBlock />\n        <Spacer direction=\"vertical\" size={'xxxl'} />\n        <StyledVerticalBlock />\n      </FlexBox>\n    ),\n    {\n      info: Readme,\n    }\n  )\n"],"file":"Spacer.story.js"}