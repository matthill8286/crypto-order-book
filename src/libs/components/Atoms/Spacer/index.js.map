{"version":3,"sources":["../../../../src/components/Atoms/Spacer/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './Spacer'\n"],"file":"index.js"}