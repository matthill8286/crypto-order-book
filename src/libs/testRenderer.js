import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { saiyanTheme } from "./styles/sc-vars-saiyan";
export function renderWithTheme(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, tree)).toJSON();
}
export function renderWithThemeAndRouter(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Router, null, tree))).toJSON();
}
export function renderWithRouter(tree) {
  return renderer.create(React.createElement(Router, null, tree)).toJSON();
}
export function mountWithTheme(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, tree));
}
export function mountWithThemeAndRouter(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, React.createElement(Router, null, tree)));
}
//# sourceMappingURL=testRenderer.js.map