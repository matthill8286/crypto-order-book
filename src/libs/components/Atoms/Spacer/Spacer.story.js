import { storiesOf } from '@storybook/react';
import React from 'react';
import { FlexBox } from "../../Helper/FlexBox";
import { styled } from "../../../styles/styled";
import { Spacer } from "./Spacer";
import Readme from "./Spacer.readme.md";
var StyledBlock = styled.div.withConfig({
  displayName: "Spacerstory__StyledBlock",
  componentId: "sc-1mmx1qq-0"
})(["height:20px;width:100%;background:red;float:left;"]);
var StyledVerticalBlock = styled.div.withConfig({
  displayName: "Spacerstory__StyledVerticalBlock",
  componentId: "sc-1mmx1qq-1"
})(["height:100px;width:20px;background:red;float:left;"]);
storiesOf('Design System/Atoms/Spacer', module).add('Default', function () {
  return React.createElement(React.Fragment, null, React.createElement(StyledBlock, null), React.createElement(Spacer, {
    size: 'md'
  }), React.createElement(StyledBlock, null), React.createElement(Spacer, {
    size: 'xxxl'
  }), React.createElement(StyledBlock, null));
}, {
  info: Readme
}).add('Vertical', function () {
  return React.createElement(FlexBox, {
    flexDirection: "row"
  }, React.createElement(StyledVerticalBlock, null), React.createElement(Spacer, {
    direction: "vertical",
    size: 'md'
  }), React.createElement(StyledVerticalBlock, null), React.createElement(Spacer, {
    direction: "vertical",
    size: 'xxxl'
  }), React.createElement(StyledVerticalBlock, null));
}, {
  info: Readme
});
//# sourceMappingURL=Spacer.story.js.map