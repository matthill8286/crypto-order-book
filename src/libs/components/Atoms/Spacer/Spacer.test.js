import React from 'react';
import { renderWithTheme } from "../../../testRenderer";
import { Spacer } from "./Spacer";
describe('Spacer', function () {
  it('renders correctly', function () {
    var tree = renderWithTheme(React.createElement(Spacer, {
      size: 'md'
    }));
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly in vertical direction', function () {
    var tree = renderWithTheme(React.createElement(Spacer, {
      direction: "vertical",
      size: 'md'
    }));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=Spacer.test.js.map