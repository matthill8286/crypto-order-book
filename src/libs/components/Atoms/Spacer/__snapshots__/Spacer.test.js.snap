// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer renders correctly 1`] = `
<div
  className="Spacer__StyledSpacer-sc-66xpbd-0 ePeiab"
  direction="horizontal"
  size="md"
/>
`;

exports[`Spacer renders correctly in vertical direction 1`] = `
<div
  className="Spacer__StyledSpacer-sc-66xpbd-0 jNSAOi"
  direction="vertical"
  size="md"
/>
`;
